Makefile

```
@@ -104,7 +104,7 @@ test: manifests generate fmt vet ## Run tests.
 
 build-within-docker: vendor
 	mkdir -p /build/cluster-api-provider-cloud-director
-	go build -ldflags "-X github.com/vmware/$(CAPVCD_IMG)/version.Version=${VERSION}" -o /build/vcloud/cluster-api-provider-cloud-director main.go
+	CGO_ENABLED=0 go build -ldflags "-s -w -X github.com/vmware/$(CAPVCD_IMG)/version.Version=${VERSION}" -o /build/vcloud/cluster-api-provider-cloud-director main.go
```

Please consider not stripping the debug symbols from the executable.

If you look at the community-maintained Cluster API providers, you will see that they do not strip debug symbols from their executables.

If the symbols are available, users are able to set breakpoints in the CAPVCD code, e.g., to understand how it works, or to investigate an issue.

If the symbols are not available, users have to either use their own CAPVCD images, or build CAPVCD and use debug symbols from a separate file, which, although possible, is complicated to do.

```suggestion
	CGO_ENABLED=0 go build -ldflags "-X github.com/vmware/$(CAPVCD_IMG)/version.Version=${VERSION}" -o /build/vcloud/cluster-api-provider-cloud-director main.go
```

---

+1

---

Debug symbols more often help an attacker than a developer.  For a final release of a file, they shouldn’t be included.  Checksec, checksec.py, and other binary security tools all check for the existence of debugging symbols as a security finding.  If someone is developing or modifying CAPVCD or other cluster API providers, then removing a single flag and running docker build is a pretty simple starting point. 

I am also not looking at other community providers as an example of secure practices. Removing debugging symbols not only reduces the size greatly for the community in large but also increases the security by disabling “easy” techniques to attack binaries.

---

> If a user can set a breakpoints, then so can any attacker.

That's true. But to set a breakpoint (assuming Linux nodes), you need the SYS_PTRACE capability. Cluster API core controller Pods drop all capabilities in their securityContexts:

https://github.com/kubernetes-sigs/cluster-api/blob/c2f930f42398088825599263acc04ae559938069/config/manager/manager.yaml#L55-L57

Would you consider doing the same for CAPVCD?

https://github.com/vmware/cluster-api-provider-cloud-director/blob/a64705555e88d16718ec7bd8611efd821535100f/config/manager/manager.yaml#L-L28

---

> Removing debugging symbols ... increases the security by disabling “easy” techniques to attack binaries.

I am aware that including debugging symbols facilitates reverse engineering.

Is that a problem for CAPVCD, whose source code is public?

Besides that reason, I am not aware of others, even after doing some searching. If you can point me to a place where I can learn more, I'd appreciate it! :pray: